cmp/grid/impl/ColumnWidthCalculator.js

```diff
@@ -87,26 +84,42 @@ export class ColumnWidthCalculator {
         } catch (e) {
             console.warn(`Error calculating max data width for column "${column.colId}".`, e);
         } finally {
-            this.setCellElActive(false);
+            this.resetClassNames();
         }
     }
 
     calcLevelWidth(gridModel, records, column, options, indentationPx = 0) {
         const {field, getValueFn, renderer} = column,
+            {sizingMode} = gridModel,
             {bufferPx} = options;
 
         // 1) Get unique values
-        const values = new Set();
+        const values = new Set(),
+            valueClassMap = {};
+
         records.forEach(record => {
             if (!record) return;
             const ctx = {record, field, column, gridModel, store: gridModel.store},
                 rawValue = getValueFn(ctx),
-                value = renderer ? renderer(rawValue, ctx) : rawValue;
+                value = renderer ? renderer(rawValue, ctx) : rawValue,
+                cellClass = this.getCellClass(gridModel, column, record, rawValue),
+                rowClass = this.getRowClass(gridModel, record);
+
             values.add(value);
+
+            // 1b) Track combinations of cell and row classes applied to this value.
+            if (!valueClassMap[value]) {
```

Comment:
might be more straightforward (and efficient) to look this up once in to a variable, rather than looking up three times.